patches/components-permissions-request_type.cc.patch

```diff
@@ -10,3 +10,11 @@ index 21bce2a736087d913c693d7a73b04dd95d3e535f..b6b03ea46a8e8353c64fd92f768d34f8
    }
    NOTREACHED();
    return gfx::kNoneIcon;
+@@ -229,6 +230,7 @@ const char* PermissionKeyForRequestType(permissions::RequestType request_type) {
+     case permissions::RequestType::kWindowPlacement:
+       return "window_placement";
+ #endif
++    BRAVE_PERMISSION_KEY_FOR_REQUEST_TYPE
```

Comment:
we can't do a chomium_src override for `PermissionKeyForRequestType`?

Reply:
It would be easier if this was not a `switch` statement, as we would just do something like this in the override file:
```c++
const char* PermissionKeyForRequestType(permissions::RequestType request_type) {
#if !defined(OS_ANDROID) && !defined(OS_IOS)
  if (request_type == permissions::RequestType::kWidevine)
    return "widevine";
  return PermissionKeyForRequestType_ChromiumImpl(request_type);
}
```

But the problem is that we still need to patch this file (I think) not to get a build error because of the `kWidevine` element that we're adding to the `permissions::RequestType` enumeration:
```
error: enumeration value 'kWidevine' not handled in switch [-Werror,-Wswitch]
```

Alternatively, we could entirely override the `PermissionKeyForRequestType()` function and include the entire `switch` statement in there, as it's done already in other overrides. @bridiver Would that be a better option?

Reply:
we have that in other overrides? Do you have links? I guess this is fine as-is to avoid the error

Reply:
Sorry, I think I mixed things with what we do for redefining some constants, where we have big .cc files in chromium_src redefining many constants from upstream, such as https://github.com/brave/brave-core/blob/master/chromium_src/chrome/common/url_constants.cc for instance.

But I did a quick search now and I don't see any case were we do this for `switch` statements other than in the cases of  `brave/chromium_src/chrome/browser/shell_integration.cc` and `brave/chromium_src/base/mac/foundation_util.mm` (where it makes sense), so I take that back.

I'll leave this as-is then, thanks.